data/wp/wp-content/plugins/epfl-courses-se/include/functions.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+require_once('xml_functions.php');
+require_once('db_functions.php');
+
+/*
+$csv_file = "../csv/courses_data.csv";
+
+//Update courses from ISA and create csv data file
+updateCoursesByYearSection('2018-2019','SHS',$csv_file);
+#print_r(getTeachers());
+*/
+
+function updateCoursesFromISAByYearSection($year, $section,$csv_path){
+	
+	/************************************
+	* CREATE CSV FILE WITH COURSES DATA *
+	*************************************/
+	
+	//Init csv file with data
+	$file = fopen($csv_path."courses_data.csv",'w') or die('Could not create csv file --> '.$csv_path."courses_data.csv");
```

Comment:
Il faut trouver un autre moyen de faire les choses que d'utiliser des fichiers CSV.
D'une part Apache n'a pas de droits pour écrire sur le système de fichiers et d'une autre part, vous utilisez déjà une DB donc vous pourriez très bien mettre les données en question dans la DB.

Reply:
Le but du fichier csv est de faciliter la mise à jour des données. Le processus est le suivant :

Reply:
Alors je pense que la génération d'un CSV à la volée serait plus pertinente mais avant que vous vous lanciez là-dedans, il faut que je regarde avec l'équipe sécurité si le fait de pouvoir uploader un fichier au sein de votre plugin serait accepté ou pas. Si ce n'est pas le cas, il faudra voir pour développer le nécessaire pour éditer les données depuis la console d'administration du site web.

Reply:
Je viens d'aller voir la personne de la sécurité en lui expliquant vos besoins avec ce que vous m'avez dit et elle a dit qu'elle allait vous appeler pour bien comprendre et vous dire ce qui était possible ou pas.

Reply:
Nouvelle update. La personne de la sécurité est passée me voir et les explications que vous lui avez données par téléphone lui ont permis de mieux comprendre.
Donc, pour la suite:
- C'est ok pour l'utilisation de fichiers CSV mais ceux-ci devront être traités à la volée (upload/download) vu que pas possible d'écrire sur disque
- Il faudra idéalement faire un contrôle des fichiers CSV histoire d'être sûr de ne pas écrire n'importe quoi dans la base de données, sous-entendu du code JavaScript/HTML qui pourrait ensuite être repris pour être affiché et créerait une "porte d'entrée" pour autre chose ou encore modifierait l'affichage. Je vous laisse regarder quels checks vous voulez faire à ce niveau.

Reply:
Une autre question me vient du coup : étant donné qu'il n'y a pas d'accès en écriture dans le répertoire du plugin, le fichier log que je crée ne peux non-plus pas être créé ?
Du coup comment procéder pour créer un fichier log pour le plugin ?

`//Create log file
	$log = fopen($csv_path."update_data.log",'w') or die ('Could not create csv file --> '.$csv_path."update_data.log");`

Sur mes installation de wordpress je n'ai aucun soucis d'écriture dans le répertoire du plugin depuis le plugin, mais je ne connais pas les droits qui sont mis en place sur le site wordpress de l'EPFL, le pourquoi de ma question.

Reply:
Si vous avez installé votre WordPress sur un Windows, le système de fichier étant par défaut accessible à tout les utilisateurs, il n'y a pas de problème pour écrire le fichier log.
Dans le cas d'un Linux, avec les droits POSIX, il se peut que ça soit déjà peu plus restrictif mais je ne connais pas assez pour le dire.
Dans notre cas, seul l'utilisateur root peut écrire dans le système de fichiers (à l'exception du dossier contenant les médias car il ne dispose pas des mêmes droits).
Si vous désirez un log des actions effectuées, vous pouvez également créer une table dans la base de donnée et y écrire votre log.

Reply:
La gestion du fichier CSV a été modifiée et se fait maintenant à la volée. Le fichier log a été enlevée car pas très pertinente pour l'instant.